### ISSUE TITLE
idm_warmepumpe_warmwasserladung_ausschalttemperatur entity no longer available, and idm_warmepumpe_warmwasser_solltemperatur shows wrong value

### ISSUE THREAD

**Issue by AndyNew2** [kodebach/hacs-idm-heatpump#67]
In previous version there were two entities supported

idm_warmepumpe_warmwasser_solltemperatur and 
idm_warmepumpe_warmwasserladung_ausschalttemperatur

Actually they showed the same value. However the ...ausschalttemperatur shows always "Unavailable" and the 
idm_warmepumpe_warmwasser_solltemperatur shows a wrong outdated value, I changed long time ago.
Seems somehow broken...
Is there a new entity replacing them?

**Comment by kodebach** [comment]
Hm, my Home Assistant still shows these entities. Not sure what went wrong for you.

BTW `Warmwasser-Solltemperatur` in Home Assistant should correspond to "Zapftemperatur" in the IDM interface (under "Warmwasser").
`Warmwasserladung Ausschalttemperatur` and `Warmwasserladung Einschalttemperatur` should correspond to "Ausschalttemperatur" and "Einschalttemperatur" respectively (under "Zeitprogramm Warmwasser"). The `Warmwasserzapftemperatur` entity in Home Assistant seems to correspond to "B42 Warmwassertemperatur" only visible in the full sensor list in the IDM interface.

Do you see the values in the list you discovered in #69 (IDs 1030-1034)?

**Comment by AndyNew2** [comment]
Yes I see them correct in the GLT Monitor
I see
ID 1033 Warmwasserladung Einschalttemperatur 35°C
ID 1034 Warmwasserladung Ausschalttemperatur 40°C

I remember I got the values correct in a previous version of your integration. Somehow it gots corrupted. I struggle as well to delete old not longer entities. I don't know what happened here, and why it is so wired...
What had been the steps to remove unwanted entities?

**Comment by kodebach** [comment]
Yes, Home Assistant makes it really difficult to delete incorrect data, because normally you don't want to loose data...

> What had been the steps to remove unwanted entities?

There is a "Purge entities" service (*BE CAREFUL* not to delete the wrong entities). If that doesn't help, you may need to edit some JSON files or access the SQLite database. If that's the case, please look for help on the [Home Assistant forum](https://community.home-assistant.io/). There are already a few posts about deleting entities, maybe one of them is what you need.

The entity list in the settings should also mark any restored entities, which you can delete via the UI and the statistics tab in the Home Assistant developer tools ("Entwicklerwerkzeuge") shows a "Fix issues" ("Problem beheben") button for all entities which issues.

Regarding this integration specifically: AFAIK there should be no need to delete anything, as long as you don't care about the entity IDs. Also the entities in question here were not affected by any updates.

**Comment by AndyNew2** [comment]
Found the issue, you are right, you not changed that bit of code, but I changed the target value for the "Ausschalttemperatur".
Your integration wants to see this value between 46 min and 53 max. That was the issue, since I lowert the value to 43 and that made it unavailable. Not checked the log. My IDM GUI states, this value can be betwen 35 to 55°C. Therefore I propose you change the file:
sensor_addresses.py in line 660 to 35 and line 661 to 55
I think this are the aligned values.

To be honest I am not a fan of checking again the values from the source. Of course if they are totally rubbish, they should be prevented, but I would open the limits to something imaginable to be valid. So instead of taking the borders from IDM (GUI), which I suspect to change from version to version, just limit it to a reasonable wider range, which would be still possible.
So for hotwater I would allow any value between 5°C to 99°C and leave the rest to the heatpump to limit it further.
Of course this would be differnt for values you support writing. For writing the borders needs to be limited to the spec.
